packages/editor-ui/src/components/Node/NodeCreator/composables/useViewStacks.ts

```diff
@@ -83,16 +83,23 @@ export const useViewStacks = defineStore('nodeCreatorViewStacks', () => {
 		}
 
 		if (stack.search && searchBaseItems.value) {
-			const searchBase =
-				searchBaseItems.value.length > 0
-					? searchBaseItems.value
-					: flattenCreateElements(stack.baselineItems ?? []);
-
+			let searchBase: INodeCreateElement[] = searchBaseItems.value;
 			const canvasHasAINodes = useCanvasStore().aiNodes.length > 0;
-			const filteredNodes =
-				isAiRootView(stack) || canvasHasAINodes ? searchBase : filterOutAiNodes(searchBase);
 
-			const searchResults = extendItemsWithUUID(searchNodes(stack.search || '', filteredNodes));
+			if (searchBaseItems.value.length === 0) {
+				searchBase = flattenCreateElements(stack.baselineItems ?? []);
+			}
+
+			if (
+				// Filter-out AI sub-nodes if canvas has no AI nodes and the root view is not AI
+				!(isAiRootView(stack) || canvasHasAINodes) ||
```

Comment:
nit: Could extract `isAiRootView(stack) || canvasHasAINodes` into a new computed (also used below in `groupIfAiRootNodes`